### PR TITLE
testing: removing dependencies

### PR DESCRIPTION
**Pull request by FranciscoKurpiel**
Proteus have only test dependencies, and this is only for tests, to help with assertions. The kind of task being delegated are quite simple, like asserting that two values are equal, and the library was designed before go.18, meaning that many parameters are of type `interface{}`, resulting in less opportunities for the compiler to catch errors.

This PR removes the test library and replaces it by an internal test library. This will remove 23k LOC from the proteus repository.

The test introduces potentially breaking change on the `plog` package, where `Entry.Caller` is changed from pointer to value. Due to go's automatic type indirection, code that only makes reference to `Caller` fields will not be affected, but other things like checking for `nil` would break.

### CODE REVIEW

File: internal/assert/error.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	if err != nil {
+		return true
+	}
+
+	t.Error("Expected error, but got nil")
+	return false
+}
+
+// PanicsNow asserts that the provided value is an error, terminating
+// the test if this is not true.
```
Comment (fho) [inline internal/assert/error.go:54]: description is wrong

File: internal/assert/string.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package assert
+
+import (
+	"strings"
+	"testing"
+)
+
+// StringContains asserts that the small string contains the smaller. Returns
```
Comment (fho) [inline internal/assert/string.go:8]: which one is the smaller and which one the small one?

```suggestion
// StringContains asserts that s contains substr. Returns
```

**Review by fho — APPROVED** [review]

File: internal/assert/string.go (hunk 2 above)
Comment (FranciscoKurpiel) [reply internal/assert/string.go:8]: I renamed the variables to match the names used on `strings.Contains and missed the godocs. Documentation is updated.

File: internal/assert/error.go (hunk 1 above)
Comment (FranciscoKurpiel) [reply internal/assert/error.go]: Fixed.

**Pull request merged**